src/lib/components/Avatar/index.tsx

```diff
@@ -1,97 +1,74 @@
 import classNames from 'classnames';
-import { FC, PropsWithChildren } from 'react';
+import { ComponentProps, FC, PropsWithChildren } from 'react';
+import { Sizes } from '../Flowbite/FlowbiteTheme';
+import { useTheme } from '../Flowbite/ThemeContext';
 import AvatarGroup from './AvatarGroup';
 import AvatarGroupCounter from './AvatarGroupCounter';
 
-export type AvatarProps = PropsWithChildren<{
+export interface AvatarProps extends PropsWithChildren<ComponentProps<'div'>> {
   alt?: string;
-  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
-  rounded?: boolean;
   bordered?: boolean;
   img?: string;
-  status?: 'offline' | 'online' | 'away' | 'busy';
-  statusPosition?: 'top-left' | 'top-right' | 'bottom-right' | 'bottom-left';
+  rounded?: boolean;
+  size?: keyof Sizes;
   stacked?: boolean;
-}>;
-
-const sizeClasses: Record<AvatarProps['size'] & string, string> = {
-  xs: 'w-6 h-6',
-  sm: 'w-8 h-8',
-  md: 'w-10 h-10',
-  lg: 'w-20 h-20',
-  xl: 'w-36 h-36',
-};
-
-const statusClasses: Record<AvatarProps['status'] & string, string> = {
-  offline: 'bg-gray-400',
-  online: 'bg-green-400',
-  away: 'bg-yellow-400',
-  busy: 'bg-red-400',
-};
-
-const statusPositionClasses: Record<AvatarProps['statusPosition'] & string, string> = {
-  'top-right': '-top-1 -right-1',
-  'top-left': '-top-1 -left-1',
-  'bottom-left': '-bottom-1 -left-1',
-  'bottom-right': '-bottom-1 -right-1',
-};
+  status?: 'away' | 'busy' | 'offline' | 'online';
+  statusPosition?: 'bottom-left' | 'bottom-right' | 'top-left' | 'top-right';
+}
 
 const AvatarComponent: FC<AvatarProps> = ({
   alt = '',
-  img,
-  status,
+  bordered = false,
   children,
-  statusPosition = 'top-left',
-  size = 'md',
+  img,
   rounded = false,
-  bordered = false,
+  size = 'md',
   stacked = false,
+  status,
+  statusPosition = 'top-left',
+  ...props
 }) => {
+  const theme = useTheme().theme.avatar;
+
   return (
-    <div data-testid="avatar-element" className="flex items-center space-x-4">
+    <div className={theme.base} data-testid="avatar-element" {...props}>
```

Comment:
why do we need `{...props}`?

Reply:
By any chance that it allow users to access the `className` like this? I would like to avoid acess to `className` directly.